src/chain.cpp

```diff
@@ -146,7 +146,7 @@ int64_t GetBlockProofEquivalentTime(const CBlockIndex& to, const CBlockIndex& fr
     if (r.bits() > 63) {
         return sign * std::numeric_limits<int64_t>::max();
     }
-    return sign * r.GetLow64();
+    return sign * static_cast<int64_t>(r.GetLow64());
```

Comment:
Correct me if I'm wrong, but if `GetLow64()` result is greater than 9223372036854775807 then the cast result is negative?

Reply:
It can't be, due to the `(r.bits() > 63)` above.

Reply:
Ah right, and it's like 2 lines above... Sorry.